README.md

```diff
@@ -19,15 +19,18 @@ dependencies {
 
 The wrapper can make use of one of the following HTTP clients: 
 
+- [HttpURLConnection](https://docs.oracle.com/javase/8/docs/api/java/net/HttpURLConnection.html) (default)
+- [OKHttp](https://github.com/square/okhttp)
+    - `compile 'com.squareup.okhttp3:okhttp:VERSION'`
 - `Spring Web`
-    - Dependency included by default
+    - `compile 'org.springframework:spring-web:VERSION'`
 - [Apache Http Components HttpClient](https://hc.apache.org/)
-    - Gradle dependency: `compile 'org.apache.httpcomponents:httpclient:VERSION'`
+    - `compile 'org.apache.httpcomponents:httpclient:VERSION'`
```

Comment:
I am starting to think it may be time to split all these implementations into separate Gradle modules.
this will require moving out integration tests from the main module to a separate module as well because it would need to depend on all clients (since integration tests use all clients).
having multiple modules would also be helpful if we cannot afford to support some older clients due to time constraints -  it would be easy enough to remove a module from the build.

Reply:
Sounds very good, but I think that suggestion kind of out of scope of this PR.